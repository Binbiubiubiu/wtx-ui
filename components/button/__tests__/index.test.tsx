import React from 'react';
import { render, shallow } from 'enzyme';
import { Button } from '..';
import toJson from 'enzyme-to-json';

describe('<Button>', () => {
  it('should render <Button> components', () => {
    const wrapper = render(
      <div>
        <Button>默认视频</Button>
        <Button up>上一页</Button>
        <Button down>下一页</Button>
        <Button disabled>禁用状态</Button>
        <Button actived>激活状态</Button>
      </div>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should style output is not null', () => {
    expect(require('../style')).not.toBeNull();
  });

  it('should button has arrow', () => {
    const wrapper = render(
      <div>
        <Button up>上一页</Button>
        <Button down>下一页</Button>
      </div>,
    );

    expect(wrapper.find('.wtx-button__up')).toHaveLength(1);
    expect(wrapper.find('.wtx-button__down')).toHaveLength(1);
  });

  it('should button can be click', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Button onClick={onClick}>可点击</Button>);
    wrapper.find('button').simulate('click');
    expect(onClick).toHaveBeenCalled();
  });

  it("should button disabled can't be click", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Button disabled onClick={onClick}>
        禁用状态
      </Button>,
    );
    wrapper.find('button').simulate('click');
    expect(onClick).not.toHaveBeenCalled();
  });

  it('should button can be actived', () => {
    const wrapper = render(<Button actived>激活状态</Button>);
    expect(wrapper.hasClass('wtx-button__actived')).toBeTruthy();
  });
});
