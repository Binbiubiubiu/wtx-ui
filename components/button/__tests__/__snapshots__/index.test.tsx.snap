// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> should render <Button> components 1`] = `
<div>
  <button
    class="wtx-button"
    type="button"
  >
    默认视频
  </button>
  <button
    class="wtx-button wtx-button__up"
    type="button"
  >
    上一页
  </button>
  <button
    class="wtx-button wtx-button__down"
    type="button"
  >
    下一页
  </button>
  <button
    class="wtx-button"
    disabled=""
    type="button"
  >
    禁用状态
  </button>
  <button
    class="wtx-button wtx-button__actived"
    type="button"
  >
    激活状态
  </button>
</div>
`;
